### PR TITLE
[Fix] Favorite and NaN

### PR DESCRIPTION
**Pull request by JulesGuesnon**
# Description
This PR fixes the favorite contract and the proposal durations inputs:
- Now if you refresh you should have the same contract selected than before
- If you enter a wrong input you should have an error message

### CODE REVIEW

File: components/signersForm.tsx [hunk 1]
```diff
@@ -191,12 +192,17 @@ const SignersForm: FC<{
         if (values.requiredSignatures > values.validators.length) {
           errors.requiredSignatures = `threshold too high. required number of signatures: ${values.requiredSignatures}, total amount of signers: ${values.validators.length}`;
         }
+
+        if (isNaN(parseInt(values.effectivePeriod as any))) {
```
Comment (rueshyna) [inline components/signersForm.tsx:196]: It seems to be a JS thing that `parseInt("12a0test")` will be `12` ?
Maybe use another way to check number? 
(I personally recommend not use `any` if possible)

**Review by rueshyna — COMMENTED** [review]
It seems that some buttons can't be clicked in this branch
- save button from setting
- continue button of second step from new/import wallet
The above are just the one I checked.

**Comment by JulesGuesnon** [comment]
It should be fixed now

File: components/create/aliases.tsx [hunk 2]
```diff
@@ -91,12 +92,19 @@ function Aliases() {
           }
           return err;
         });
+
+        if (isNaN(Number(values.effectivePeriod))) {
```
Comment (rueshyna) [inline components/create/aliases.tsx:96]: I forgot to say that would be nice to check if number `>0`.

File: components/signersForm.tsx [hunk 3]
```diff
@@ -191,6 +192,12 @@ const SignersForm: FC<{
         if (values.requiredSignatures > values.validators.length) {
           errors.requiredSignatures = `threshold too high. required number of signatures: ${values.requiredSignatures}, total amount of signers: ${values.validators.length}`;
         }
+
+        if (isNaN(Number(values.effectivePeriod))) {
```
Comment (rueshyna) [inline components/signersForm.tsx:196]: also here

**Review by rueshyna — COMMENTED** [review]
continue button of second step from import wallet can't click, the others looks good.

File: components/create/aliases.tsx [hunk 4]
```diff
@@ -93,7 +93,8 @@ function Aliases() {
           return err;
         });
 
-        if (isNaN(Number(values.effectivePeriod))) {
+        const parsedNumber = Number(values.effectivePeriod);
+        if (isNaN(parsedNumber) || parsedNumber <= 0) {
```
Comment (rueshyna) [inline components/create/aliases.tsx:97]: Sorry, not equal `0`. it doesn't make sense to have 0 duration and contract part will throw error for 0 duration.

File: components/signersForm.tsx [hunk 5]
```diff
@@ -193,7 +193,8 @@ const SignersForm: FC<{
           errors.requiredSignatures = `threshold too high. required number of signatures: ${values.requiredSignatures}, total amount of signers: ${values.validators.length}`;
         }
 
-        if (isNaN(Number(values.effectivePeriod))) {
+        const parsedNumber = Number(values.effectivePeriod);
+        if (isNaN(parsedNumber) || parsedNumber <= 0) {
```
Comment (rueshyna) [inline components/signersForm.tsx:198]: the same

Comment (rueshyna) [reply components/signersForm.tsx:198]: my bad. please ignore it.

**Review by rueshyna — APPROVED** [review]

**Pull request merged**